cli/cli.go

```diff
@@ -2,43 +2,105 @@ package cli
 
 import (
 	"fmt"
-	"github.com/shpota/goxygen/codegen"
 	"io"
 	"regexp"
+	"strings"
+
+	"github.com/shpota/goxygen/codegen"
 )
 
 func Start(w io.Writer, commands []string) {
-	if len(commands) == 1 && commands[0] == "help" {
+	argsLen := len(commands)
+	if argsLen == 1 && commands[0] == "help" {
 		fmt.Fprintln(w, usage)
 		return
 	}
-	if len(commands) == 2 && commands[0] == "init" {
-		projectName := commands[1]
-		validName, _ := regexp.MatchString("^[a-zA-Z0-9_-]+$", projectName)
-		if validName {
-			codegen.Generate(projectName)
+	if (argsLen == 2 || argsLen == 3) && commands[0] == "init" {
+		var frontendTech string
+		var projectName string
+
+		projectName, ok, errMsg := extractProjectName(commands[argsLen-1])
+		if !ok {
+			fmt.Fprintln(w, errMsg)
 			return
 		}
-		fmt.Fprintln(w, invalidName)
+		if argsLen == 2 {
```

Comment:
Would it not be more clear if you put this block one level up?

```
if argsLen == 2 && commands[0] == "init" { ... }
```
What do you think?